### PR TITLE
filter user list by account type

### PR DESCRIPTION
**Pull request by meleu**
Added options to filter the User List by account type, like in the image below:
userlistfilters

### CODE REVIEW

File: lib/database/user.php [hunk 1]
```diff
@@ -1484,53 +1484,62 @@ function getControlPanelUserInfo( $user, &$libraryOut )
...
+function getUserListByPerms( $sortBy, $offset, $count, &$dataOut, $requestedBy , &$perms = NULL)
 {
     settype( $offset, 'integer' );
     settype( $count, 'integer' );
 
+    $permsFilter = NULL;
+    if( $perms != NULL)
+    {
+        settype( $perms, 'integer' );
+        if( $perms >= -2 && $perms <= 4 ) // details about the numbers in src/Permissions.php
```
Comment (ScottFromDerby) [inline lib/database/user.php]: could this be 
$perms >= \RA\Permissions::Spam && $perms <= \RA\Permissions::Admin
instead?

Also is there a benefit in allowing people to see Spam accounts? There may be, for other admin at least

File: public/userList.php [hunk 2]
```diff
@@ -30,12 +36,34 @@
...
             echo "<h2 class='longheader'>User List:</h2>";
+
+            echo "<p>Filter: ";
+
+            if( $permissionName == NULL )
+                echo "<b>All Users</b>";
+            else
+                echo "<a href='/userList.php?s=$sortBy'>All Users</a>";
+
+            for( $i = -2; $i <= 4; $i++ ) // details about the numbers in src/Permissions.php
```
Comment (ScottFromDerby) [inline public/userList.php]: Same here for magic numbers

File: lib/database/user.php (hunk 1 above)
Comment (meleu) [reply lib/database/user.php]: >  is there a benefit in allowing people to see Spam accounts?

Well, not really sure about that. I'm just curious to see how many of them exist... I'm glad to remove this filter if it makes this PR more acceptable :sweat_smile:

**Comment by meleu** [comment]
@ScottFromDerby could you make these changes alive in the server so we can ~~try to break them~~ see how they look like?

**Pull request merged**